openpype/hosts/houdini/api/lib.py

```diff
@@ -747,3 +751,97 @@ def get_camera_from_container(container):
 
     assert len(cameras) == 1, "Camera instance must have only one camera"
     return cameras[0]
+
+
+def get_context_var_changes():
+    """get context var changes."""
+
+    houdini_vars_to_update = {}
+
+    project_settings = get_current_project_settings()
+    houdini_vars_settings = \
+        project_settings["houdini"]["general"]["update_houdini_var_context"]
+
+    if not houdini_vars_settings["enabled"]:
+        return houdini_vars_to_update
+
+    houdini_vars = houdini_vars_settings["houdini_vars"]
+
+    # No vars specified - nothing to do
+    if not houdini_vars:
+        return houdini_vars_to_update
+
+    # Get Template data
+    template_data = get_current_context_template_data()
+
+    # Set Houdini Vars
+    for item in houdini_vars:
+        # For consistency reasons we always force all vars to be uppercase
+        # Also remove any leading, and trailing whitespaces.
+        var = item["var"].strip().upper()
+
+        # get and resolve template in value
+        item_value = StringTemplate.format_template(
+            item["value"],
+            template_data
+        )
+
+        if var == "JOB" and item_value == "":
+            # sync $JOB to $HIP if $JOB is empty
+            item_value = os.environ["HIP"]
+
+        if item["is_directory"]:
+            item_value = item_value.replace("\\", "/")
+
+        current_value = hou.hscript("echo -n `${}`".format(var))[0]
+
+        if current_value != item_value:
+            houdini_vars_to_update[var] = (
+                current_value, item_value, item["is_directory"]
+            )
+
+    return houdini_vars_to_update
+
+
+def update_houdini_vars_context():
+    """Update asset context variables"""
+
+    for var, (_old, new, is_directory) in get_context_var_changes().items():
+        if is_directory:
+            try:
+                os.makedirs(new)
+            except OSError as e:
+                if e.errno != errno.EEXIST:
+                    print(
+                        "Failed to create ${} dir. Maybe due to "
+                        "insufficient permissions.".format(var)
+                    )
+
+        hou.hscript("set {}={}".format(var, new))
+        os.environ[var] = new
+        print("Updated ${} to {}".format(var, new))
+
+
+def update_houdini_vars_context_dialog():
+    """Show pop-up to update asset context variables"""
+    update_vars = get_context_var_changes()
+    if not update_vars:
+        # Nothing to change
+        print("Nothing to change, Houdini vars are already up to date.")
+        return
+
+    message = "\n".join(
+        "${}: {} -> {}".format(var, old or "None", new)
+        for var, (old, new, _is_directory) in update_vars.items()
+    )
+    parent = hou.ui.mainQtWindow()
+    dialog = popup.PopupUpdateKeys(parent=parent)
```

Comment:
```suggestion
    dialog = popup.Popup(parent=parent)
```
Can you test whether this would work too?
It should show the same pop-up but without the Update Keys checkbox :)

Reply:
![image](https://github.com/ynput/OpenPype/assets/20871534/08db4a66-f20a-4894-8bdc-4887f0f586d5)

Reply:
Ah I see, change `on_clicked_state` to `on_clicked`

Reply:
https://github.com/ynput/OpenPype/pull/5651#discussion_r1348444804

That just looks like your "new" values are empty strings? :)

Reply:
![image](https://github.com/ynput/OpenPype/assets/20871534/672005e3-75b9-499c-aaaa-8756fa9a93da)
![image](https://github.com/ynput/OpenPype/assets/20871534/33021822-8787-4fe0-81bb-f4a85c01c322)

Reply:
Seems better at least without the checkbox we we weren't actually using - right?
I mean it's far from beautiful, but fine prototype.

Reply:
okay, Then I will just add a TODO : Use better popup UI